src/pages/Tap.vue

```diff
@@ -83,25 +83,39 @@ export default {
                     function(response) {
 
                         // Add error message for invalid and fail codes.
-                        if (
-                            response.data.invalid.length > 0
-                        ) {
-                            this.showFail();
-                            this.errorMessage = "Please enter a valid voucher code.";
-                        } else if (
-                            response.data.fail.length > 0
-                        ) {
+                        var data = response.data;
+
+                        if (data.invalid.length + data.fail.length == 1) {
```

Comment:
This needs a refactor. Scan Tap and Upload will potentially use this same system.